monitor/CMakeLists.txt

```diff
@@ -8,7 +9,29 @@ if (LINUX)
 
     set(OS_MAIN_SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/main.cc)
 
-    find_package(glfw3 REQUIRED)
+    if (SKBUILD)
+        # Build GLFW for source
+        # ---- GLFW options (must be set BEFORE FetchContent_MakeAvailable) ----
+        set(GLFW_BUILD_NULL       ON  CACHE BOOL "" FORCE)
+        set(GLFW_BUILD_X11        ON  CACHE BOOL "" FORCE)
+        set(GLFW_BUILD_WAYLAND    ON CACHE BOOL "" FORCE)
+        set(GLFW_BUILD_EXAMPLES   OFF CACHE BOOL "" FORCE)
+        set(GLFW_BUILD_TESTS      OFF CACHE BOOL "" FORCE)
+        set(GLFW_BUILD_DOCS       OFF CACHE BOOL "" FORCE)
+        set(GLFW_INSTALL       OFF CACHE BOOL "" FORCE)
+        set(GLFW_LIBRARY_TYPE     "STATIC"  CACHE STRING "" FORCE)
+        set(CMAKE_POSITION_INDEPENDENT_CODE ON CACHE BOOL "" FORCE)
+
+        include(FetchContent)
+        FetchContent_Declare(glfw
+        GIT_REPOSITORY    https://github.com/glfw/glfw.git
```

Comment:
Instead of building from source, why not install via Conan and set the above options in the Conan file? That way you can find package like before? The glfw recipe supports these options https://github.com/conan-io/conan-center-index/blob/master/recipes/glfw/all/conanfile.py , that way we can upgrade the package and keep it separate in the long run.